### ISSUE TITLE
Question about minibatch size calculation when using unroll in training loop

### ISSUE THREAD

**Issue by InckieUltra** [google/brax#632]
Hi all,

I'm trying to clarify how minibatch size is calculated when using the unroll pattern in the training loop.
The following code snippets are directly based on the official brax data collection implementation.

Main loop code:

```
def f(carry, unused_t):
    current_state, current_key = carry
    current_key, next_key = jax.random.split(current_key)
    next_state, data = acting.generate_unroll(
        env,
        current_state,
        policy,
        current_key,
        unroll_length,
        extra_fields=('truncation', 'episode_metrics', 'episode_done'),
    )
    return (next_state, next_key), data

(state, _), data = jax.lax.scan(
    f,
    (state, key_generate_unroll),
    (),
    length=batch_size * num_minibatches // num_envs,
)
```
generate_unroll implementation:

```
def generate_unroll(
    env: Env,
    env_state: State,
    policy: Policy,
    key: PRNGKey,
    unroll_length: int,
    extra_fields: Sequence[str] = (),
) -> Tuple[State, Transition]:
    """Collect trajectories of given unroll_length."""

    @jax.jit
    def f(carry, unused_t):
        state, current_key = carry
        current_key, next_key = jax.random.split(current_key)
        nstate, transition = actor_step(
            env, state, policy, current_key, extra_fields=extra_fields
        )
        return (nstate, next_key), transition

    (final_state, _), data = jax.lax.scan(
        f, (env_state, key), (), length=unroll_length
    )
    return final_state, data
```
My understanding is that the shape of collected data should be
[batch_size * num_minibatches // num_envs, unroll_length, ...],
so the actual number of transitions per batch (batch size) is
batch_size * num_minibatches // num_envs * unroll_length,
and the actual number of minibatch size is
batch_size // num_envs * unroll_length.

Is this correct? Is this the effective minibatch size used for each gradient update?

Thank you for your help!
